packages/node-cli/src/core/sdksetup/SdkDownloadService.js

```diff
@@ -41,52 +45,73 @@ class SdkDownloadService {
 		const sdkDirectory = this._fileSystemService.createFolder(sdkParentDirectory, FOLDERS.NODE_CLI);
 
 		const fullURL = `${SdkProperties.getDownloadURL()}/${SdkProperties.getSdkFileName()}`;
+		const destinationFilePath = path.join(sdkDirectory, SdkProperties.getSdkFileName());
+		const proxy = process.env.SUITECLOUD_PROXY || process.env.npm_config_https_proxy || process.env.npm_config_proxy;
+		const skipProxy = SdkProperties.configFileExists();
 
 		try {
 			await executeWithSpinner({
-				action: this._downloadFile(fullURL, sdkDirectory),
+				action: this.downloadJarFilePromise(fullURL, destinationFilePath, proxy, skipProxy),
 				message: NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK, fullURL),
-			})
+			});
 			NodeConsoleLogger.info(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_SUCCESS));
-		}
-		catch (error) {
-			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)))
+		} catch (error) {
+			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)));
 		}
 	}
 
-	_downloadFile(url, sdkDirectory) {
-		const proxy = process.env.npm_config_https_proxy || process.env.npm_config_proxy;
-		const isProxyRequired = proxy && !SdkProperties.configFileExists();
-		const removeJarFilesFrom = this._removeJarFilesFrom;
+	downloadJarFilePromise(downloadUrl, destinationFilePath, proxy, skipProxy) {
```

Comment:
Could we maintain it as private?

Reply:
I was making it public because I was thinking in reusing this method from vscode-extension. But that's not clear, so I'll prefix it again with the `_` .